coalib/io/FileCache.py

```diff
@@ -0,0 +1,63 @@
+import functools
+
+__all__ = ['get_content', 'clear_cache']
+
+
+cache = {}
+
+
+def clear_cache():
+    """
+    Clear the cache entirely.
+    """
+    global cache
+    cache = {}
+
+
+def get_content(filename):
+    """
+    Get the raw contents of a file from the cache.
+    Update the cache if it doesn't contain an entry for this file already.
+    """
+    if filename in cache:
+        return cache[filename]
+    return update_cache(filename)
+
+
+def update_cache(filename):
+    """
+    Update a cache entry and return its list of lines.
+    If something's wrong, print a message, discard the cache entry,
+    and return an empty list.
+    """
+    with open(filename, 'rb') as fp:
+        contents = fp.read()
+    cache[filename] = contents
+    return contents
+
+
+class Memoize:
+    """
+    This is an edited version of the ``cached_property`` decorator given here:
```

Comment:
wait this is already a cached property? Why do we have to add `@Memoize` together with `@property`?